docs/program/apis/sessions.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Session Management with Viam's Client SDKs"
+linkTitle: "Session Management API"
+weight: 20
+type: "docs"
+description: "How to use the session management API with Viam's Client SDKs."
+tags: ["client", "sdk", "viam-server", "networking", "apis", "robot api", "session", "sessions", "session management"]
+---
+
+The session management API provides support for robot session management.
+
+A session:
+
+- allows a client to express that it is actively connected or authenticated to `viam-server` on your robot
+- supports stopping moving components when the connection is no longer active.
+
+### Purpose of session management
+
+When controlling a robot or fleet with Viam, you want a way to understand the presence of the clients that are communicating with and authenticated to each robot's `viam-server` through Viam's various [resource APIs](/program/apis/).
+
+The presence of these clients at one point in time is called a *session*.
+A session technically is any presence mechanism at the RDK application layer maintained by any client with `viam-server`.
+
+Session management allows for safer operation of robots that physically move.
+Specifically, without session management, controls that would be "sticky" (for example [`SetPower()` of a base](/components/base/#setpower)) based on the last input of a **client**, meaning the API request from the client sets the flow of electricity on a part of the robot and then does not time out, could lead to **having a robot trying to continue doing what it was told to do forever.**
+
+You do not want that to happen.
+
+#### Clients and Viam's client SDKs
+
+*Client* has multiple meanings for a Viam robot.
+Essentially, it's anything that is receiving the information served by `viam-server` running on the robot.
+
+A **client** can be an SDK script controlling the robot, an input controller, or just the different resources on the robot talking amongst themselves.
+For example, if you use Viam's module registry to [add modular resources to your robot](/extend/modular-resources/), the clients of your robot in its lifetime will include the "model servers" you instantiate on your robot for individual resources.
+
+Viam's session management API is your built-in solution to this.
+Your client maintains the session, telling the `viam-server` instance that it is still present every so often, or staying within the heartbeat window.
```

Comment:
You can adjust the heartbeat window via the robot config (needs to be done through raw JSON at the moment):

```json
  ...
  "network": {
    "sessions": {
      "heartbeat_window": "30s"
    }
  },
  ...
```

The default heartbeat window is 2 seconds if this configuration is omitted.

Reply:
Would it be correct to say
```suggestion
Each client maintains a session, telling the `viam-server` instance that it is still present every so often, or staying within the heartbeat window.
```
The page just talked about how there are multiple meanings of client. So, do all of them maintain sessions or just one?

Reply:
Just one

e.g. the python sdk maintains one session - it does *not* create a new session for each component/service client it manages

Reply:
^ I think this is consistent with current proposed docs so no changed needed?